app/screens/contacts/Contacts.tsx

```diff
@@ -259,11 +272,17 @@ const Contacts = ({ history }: RouteComponentProps) => {
   );
   const [selectedSorting, setSelectedSorting] = useState(0);
   const [isNewContactCreated, setIsNewContactCreated] = useState(false);
-  const [shouldShowPasswordModal, setShouldShowPasswordModal] = useState(false);
-  const [contactForDelete, setContactForDelete] = useState({
-    address: '',
-    nickname: '',
-  });
+  const [
+    shouldShowPasswordModal,
+    setShouldShowPasswordModal,
+  ] = useState<boolean>(false);
+
+  const [contactEditing, setContactEditing] = useState<Contact | null>(null);
+
+  const [submitFunction, setSubmitFunction] = useState<
```

Comment:
Seems flaky and non-optimal:
1. I don't see a reason to recreate the handler every time.
2. We can easily make a mistake then and try to call `null` instead of function. So for such cases better to use noop (`() => {}`) instead of  `null`.

SO I propose to:
1. Replace `shouldShowPasswordModal` with `contactToDelete: Contact | null`
4. Get rid of `setSubmitFunction` and just have a callback: `const handleDelete = ({ password }) => contactToDelete && deleteContact({ contact: contactToDelete, password });`

Reply:
Also did an investigation, but failed. Thanks.